tests/e2e/e2e_ibc_memo_test.go

```diff
@@ -0,0 +1,84 @@
+package e2e
+
+import (
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+	"gotest.tools/v3/assert"
+
+	"github.com/umee-network/umee/v6/tests/accs"
+	setup "github.com/umee-network/umee/v6/tests/e2e/setup"
+	"github.com/umee-network/umee/v6/tests/tsdk"
+	ltypes "github.com/umee-network/umee/v6/x/leverage/types"
+	"github.com/umee-network/umee/v6/x/uibc"
+)
+
+func (s *E2ETest) testIBCTokenTransferWithMemo(umeeAPIEndpoint string, atomQuota sdk.Coin) {
+	totalSupply, err := s.QueryTotalSupply(umeeAPIEndpoint)
+	s.T().Logf("total supply : %s", totalSupply.String())
+	prevIBCAtomBalance := totalSupply.AmountOf(uatomIBCHash)
+	s.T().Logf("total balance of IBC ATOM : %s", prevIBCAtomBalance.String())
+
+	//<<<< Valid MEMO  : gaia -> umee >>
+	atomFromGaia := mulCoin(atomQuota, "5.0")
+	atomFromGaia.Denom = "uatom"
+
+	atomIBCDenom := atomFromGaia
+	atomIBCDenom.Denom = uatomIBCHash
+	cdc := tsdk.NewCodec(uibc.RegisterInterfaces, ltypes.RegisterInterfaces)
+
+	// INVALID MEMO : with fallback_addr
+	//  Collteralize msg is not supported
+	msgCollateralize := []sdk.Msg{
+		ltypes.NewMsgCollateralize(accs.Alice, atomIBCDenom),
+	}
+	anyMsgOfCollateralize, err := tx.SetMsgs(msgCollateralize)
+	assert.NilError(s.T(), err)
```

Comment:
please create helper variable (at the top of the function), to make the assert calls shorter
```
assert := assert.New(s.T())
```